libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -21,23 +23,49 @@ internal class MapboxDirectionsSession(
     private val routesObservers = CopyOnWriteArraySet<RoutesObserver>()
     private var primaryRouteOptions: RouteOptions? = null
 
+    private var routesAndOriginWrapper: RoutesAndOriginWrapper? = null
```

Comment:
it looks like there are multiple sources of truth for the `routerOrigin`. It's in both the RoutesAndOriginWrapper and a stand alone val

did we think about making this a single object?

Reply:
multiple sources, but one is private another is public. And it still has a single root. 
```kotlin
override val routerOrigin: RouterOrigin?
        get() = routesAndOriginWrapper?.origin
```
Did you mean that we might make `public val routesAndOriginWrapper` and restrict `routerOrigin` and `routerOrigin`?